### ISSUE TITLE
[Bug]: Autodelete watched videos failure on new installations without any watched videos

### ISSUE THREAD

**Issue by derekslenk** [tubearchivist/tubearchivist#592]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Ubuntu 23.10 via Docker Compose

## Describe the bug
Comments failed to download after download queue finished completing, but was actually caused by an ES index issue for deleting auto-delete videos

## Steps To Reproduce
1. Fresh docker-compose install
2. Subscribe to channel and set an autodelete after x days.
3. Rescan subs and start download.
4. Hit stop or wait for queue to finish, should experience error.

## Expected behavior
Comments download, and autodelete works

### Relevant log output

```shell
tubearchivist    | [2023-11-10 17:38:50,257: WARNING/ForkPoolWorker-8] ZHaDFrygh4w: Downloading video
tubearchivist    | [task][123fdf94-5394-487e-a6de-726ac8cda0a4]: received STOP signal.
tubearchivist    | [2023-11-10 17:40:58,513: WARNING/ForkPoolWorker-8] ZHaDFrygh4w: get metadata from youtube
tubearchivist    | [2023-11-10 17:41:00,700: WARNING/ForkPoolWorker-8] UCYF1kiTFCcYPqGTqXKt3JJg: get metadata from es
tubearchivist    | [2023-11-10 17:41:00,932: WARNING/ForkPoolWorker-8] ZHaDFrygh4w: get ryd stats
tubearchivist    | [2023-11-10 17:41:01,126: WARNING/ForkPoolWorker-8] ZHaDFrygh4w: get sponsorblock timestamps
tubearchivist    | [2023-11-10 17:41:01,388: WARNING/ForkPoolWorker-8] ZHaDFrygh4w: sponsorblock failed: 404
tubearchivist    | [2023-11-10 17:41:01,390: WARNING/ForkPoolWorker-8] ZHaDFrygh4w-en: get user uploaded subtitles
tubearchivist    | [2023-11-10 17:41:01,390: WARNING/ForkPoolWorker-8] ZHaDFrygh4w-en: get auto generated subtitles
archivist-redis  | 8:M 10 Nov 2023 22:41:04.029 * 100 changes in 300 seconds. Saving...
archivist-redis  | 8:M 10 Nov 2023 22:41:04.031 * Background saving started by pid 17
archivist-redis  | 17:C 10 Nov 2023 22:41:04.052 * DB saved on disk
archivist-redis  | 17:C 10 Nov 2023 22:41:04.054 * Fork CoW for RDB: current 1 MB, peak 1 MB, average 1 MB
archivist-redis  | 8:M 10 Nov 2023 22:41:04.132 * Background saving terminated with success
tubearchivist    | [2023-11-10 17:41:04,152: WARNING/ForkPoolWorker-8] UCLHsZ4X7YemjxRrvq0AI4LA: delete older than 1d
tubearchivist    | [2023-11-10 17:41:04,222: WARNING/ForkPoolWorker-8] {"error":{"root_cause":[{"type":"query_shard_exception","reason":"No mapping found for [player.watched_date] in order to sort on","index_uuid":"vQeH_yYgRECrz_VRMK7MjA","index":"ta_video"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"ta_video","node":"SRDEUo6WR8iaTm1x_9SxIA","reason":{"type":"query_shard_exception","reason":"No mapping found for [player.watched_date] in order to sort on","index_uuid":"vQeH_yYgRECrz_VRMK7MjA","index":"ta_video"}}]},"status":400}
tubearchivist    | [2023-11-10 17:41:04,240: WARNING/ForkPoolWorker-8] 123fdf94-5394-487e-a6de-726ac8cda0a4 Failed callback
tubearchivist    | [2023-11-10 17:41:04,244: ERROR/ForkPoolWorker-8] Task download_pending[123fdf94-5394-487e-a6de-726ac8cda0a4] raised unexpected: KeyError('hits')
tubearchivist    | Traceback (most recent call last):
tubearchivist    |   File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
tubearchivist    |     R = retval = fun(*args, **kwargs)
tubearchivist    |                  ^^^^^^^^^^^^^^^^^^^^
tubearchivist    |   File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 760, in __protected_call__
tubearchivist    |     return self.run(*args, **kwargs)
tubearchivist    |            ^^^^^^^^^^^^^^^^^^^^^^^^^
tubearchivist    |   File "/app/home/tasks.py", line 208, in download_pending
tubearchivist    |     videos_downloaded = downloader.run_queue(auto_only=auto_only)
tubearchivist    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tubearchivist    |   File "/app/home/src/download/yt_dlp_handler.py", line 196, in run_queue
tubearchivist    |     DownloadPostProcess(self).run()
tubearchivist    |   File "/app/home/src/download/yt_dlp_handler.py", line 42, in run
tubearchivist    |     self.auto_delete_overwrites()
tubearchivist    |   File "/app/home/src/download/yt_dlp_handler.py", line 75, in auto_delete_overwrites
tubearchivist    |     self._auto_delete_watched(data)
tubearchivist    |   File "/app/home/src/download/yt_dlp_handler.py", line 80, in _auto_delete_watched
tubearchivist    |     to_delete = IndexPaginate("ta_video", data).get_results()
tubearchivist    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tubearchivist    |   File "/app/home/src/es/connect.py", line 164, in get_results
tubearchivist    |     all_results = self.run_loop()
tubearchivist    |                   ^^^^^^^^^^^^^^^
tubearchivist    |   File "/app/home/src/es/connect.py", line 194, in run_loop
tubearchivist    |     all_hits = response["hits"]["hits"]
tubearchivist    |                ~~~~~~~~^^^^^^^^
tubearchivist    | KeyError: 'hits'
tubearchivist    | [2023-11-10 17:41:04,245: WARNING/ForkPoolWorker-8] 123fdf94-5394-487e-a6de-726ac8cda0a4 return callback
tubearchivist    | Fri Nov 10 17:41:24 2023 - SIGPIPE: writing to a closed pipe/socket/fd (probably the client disconnected) on request /static/favicon/favicon-16x16.de290f6207d0.png (ip 192.168.12.80) !!!
tubearchivist    | Fri Nov 10 17:41:24 2023 - uwsgi_response_sendfile_do(): Broken pipe [core/writer.c line 655] during GET /static/favicon/favicon-16x16.de290f6207d0.png (192.168.12.80)
tubearchivist    | OSError: write error
```

### Anything else?

https://discord.com/channels/920056098122248193/1172667528644919438

**Comment by bbilly1** [comment]
Thanks for reporting. This is missing a mapping.

Work around for that is to mark *any* video as watched, that should generate the mapping automatically and allow the query to run.

**Comment by bbilly1** [comment]
fixed with v0.4.4
